library/core/src/intrinsics.rs

```diff
@@ -2020,6 +2020,20 @@ extern "rust-intrinsic" {
     #[rustc_safe_intrinsic]
     pub fn saturating_sub<T: Copy>(a: T, b: T) -> T;
 
+    /// This is a *typed* read, `copy *p` in MIR.
+    ///
+    /// The stabilized form of this intrinsic is [`crate::ptr::read`], so
+    /// that can be implemented without needing to do an *untyped* copy
+    /// via [`copy_nonoverlapping`], and thus can get proper metadata.
```

Comment:
```suggestion
    /// The stabilized form of this intrinsic is [`crate::ptr::read`], so that
    /// it is easier for the compiler to generate a load with proper metadata.
```